### ISSUE TITLE
Suggestion: Support an OPENFAAS_PREFIX env var

### ISSUE THREAD

**Issue by rgee0** [openfaas/faas-cli#550]
Having to repeatedly supply `--prefix` was mentioned in #479 and #496 covers the broader idea of having a concept similar to the AWS CLI where profiles can be specified to save the user having to repeatedly supply details.  Should we support an env var which when present would negate the need to provide `--prefix`, in much the same way as OPENFAAS_URL negates the need to provide `--gateway/-g`

## Expected Behaviour
```
$ export  OPENFAAS_PREFIX=rgee0
$ faas-cli new --lang node newFn
$ cat newFn.yml
provider:
  name: faas
  gateway: http://127.0.0.1:8080
functions:
  fn1:
    lang: node
    handler: ./newFn
    image: rgee0/newFn:latest
```
## Current Behaviour
Provide `--prefix` on every call to new

## Possible Solution
Follow similar approach to OPENFAAS_URL with a precedence order established so that if `--prefix` is present it will over ride the env var.

## Context
Improved UX

## Your Environment

N/A

**Comment by martindekov** [comment]
I think this is a very good idea. Most of the time I am pushing to my dockerhub and it is annoying to write the prefix every time. Is it approved? If so I can start working on it

**Comment by ivanayov** [comment]
+1 for this

I wonder if it's better to make the name more descriptive, but  just`OPENFAAS_PREFIX` is also fine.
IMO  it should be taken into account only by `faas new` and prioritised after the `--prefix` flag.

**Comment by rgee0** [comment]
> How would we expect this to work if the YAML file already has a prefix?

I was anticipating a precedence order approach similar to that which we use in the `getGateway()` method.  Whether the order itself is the same as that in the gateway decision is open to discussion.

> I wonder if it's better to make the name more descriptive, but justOPENFAAS_PREFIX is also fine.

It mirrors the flag, which the user should already be familiar with.  Happy to deviate on the suggested name, but in doing so we should consider whether the flag is affected in the same way.

**Comment by martindekov** [comment]
Is there any movement on this Issue?

**Comment by alexellis** [comment]
I think it would make more sense as `OPENFAAS_USER_PREFIX` - for context - `--prefix` was only chosen for the CLI for brevity.

I think there are two problems here so let's make sure we are solving the right one first:

## 1. Users want to hack on a stack with a prefix such as `openfaas` or `rgee0` but have to manually hack on the YAML and keep that in sync

## 2. Some people find adding `--prefix` to `faas new` tedious

Alex

**Comment by burtonr** [comment]
I'll put in a vote to focus only on enabling a default for the `faas new` command. When creating new functions, it's easy to forget to add the `--prefix` to the argument list. Then, running `faas up` throws an error since it can't push the image, and causes a minor inconvenience.

A second update could be made to take into account an existing `.yml` file and possibly add the prefix if it doesn't exist, but I think that's a separate issue. If I (the user) am creating a yml file, I know to put the prefix in, but when the CLI  creates the yml file, it should add the prefix automatically using this new env var, or even better, the `.openfaas/config`.

I believe the order of precedence should be `--prefix` > `OPENFAAS_PREFIX` > `.openfaas/config`

**Comment by alexellis** [comment]
Sounds good to me @burtonr

**Comment by martindekov** [comment]
I can take this if we should go with @burtonr ‘a approach

**Comment by martindekov** [comment]
Ah this is already started ...

**Comment by alexellis** [comment]
Is it started in the way Burton mentioned? For `faas-cli new`

**Comment by rgee0** [comment]
Implemented in #634 & released in 0.8.10
